### PR TITLE
Merge 1.18.0

### PR DESCRIPTION
**Pull request by egegunes**
**CHANGE DESCRIPTION**
---
Merging changes in release branch into main.

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?
- [ ] Are OpenShift compare files changed for E2E tests (`compare/*-oc.yml`)?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are all needed new/changed options added to the [Helm Chart](https://github.com/percona/percona-helm-charts)?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported MongoDB version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

File: deploy/cr-minimal.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: PerconaServerMongoDB
 metadata:
   name: minimal-cluster
 spec:
-  crVersion: 1.18.0
+  crVersion: 1.19.0
```
Comment (eleo007) [inline deploy/cr-minimal.yaml:6]: If we are keeping 1.18.0 images in this PR, maybe let's update crVersion in next pr where we will revert images?

**Comment by eleo007** [comment]
> @ptankov could you please update upgrade-consistency tests?

If we are keeping 1.18.0 images etc, maybe let's update tests in separate pr with images?

**Comment by ptankov** [comment]
> @ptankov could you please update upgrade-consistency tests?
done

**Comment by eleo007** [comment]
I do not see changes to `version/version.go`

**Comment by ptankov** [comment]
> I do not see changes to `version/version.go`

check now

**Comment by ptankov** [comment]
> I do not see changes to `version/version.go`

`version/version.go` is as it should be - `1.19.0`, it is already updated as part of a commit by Anton Averianov on November 12, 2024 at 2:09 PM

```
K8SPSMDB-1215: fix ExternalTrafficPolicy for different service types (#1673)
fix ExternalTrafficPolicy for different service types
change tests for service check
Fix for API compatibility: set ExternalTrafficPolicy by version
Fix for API compatibility: set ExternalTrafficPolicy by version + service
update version to 1.19.0
```

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/arbiter.log) |
| balancer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/balancer.log) |
| custom-replset-name | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/custom-replset-name.log) |
| custom-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/custom-tls.log) |
| custom-users-roles | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/custom-users-roles.log) |
| custom-users-roles-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/custom-users-roles-sharded.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/cross-site-sharded.log) |
| data-at-rest-encryption | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/data-at-rest-encryption.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/data-sharded.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/demand-backup.log) |
| demand-backup-fs | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/demand-backup-fs.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/demand-backup-eks-credentials.log) |
| demand-backup-physical | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/demand-backup-physical.log) |
| demand-backup-physical-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/demand-backup-physical-sharded.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/demand-backup-sharded.log) |
| expose-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/expose-sharded.log) |
| ignore-labels-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/ignore-labels-annotations.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/init-deploy.log) |
| finalizer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/finalizer.log) |
| ldap | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/ldap.log) |
| ldap-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/ldap-tls.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/limits.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/liveness.log) |
| mongod-major-upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/mongod-major-upgrade.log) |
| mongod-major-upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/mongod-major-upgrade-sharded.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/monitoring-2-0.log) |
| multi-cluster-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/multi-cluster-service.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/non-voting.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/one-pod.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/operator-self-healing-chaos.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/pitr.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/pitr-sharded.log) |
| pitr-physical | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/pitr-physical.log) |
| pvc-resize | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/pvc-resize.log) |
| recover-no-primary | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/recover-no-primary.log) |
| replset-overrides | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/replset-overrides.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/rs-shard-migration.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/scheduled-backup.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/security-context.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/self-healing-chaos.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/service-per-pod.log) |
| serviceless-external-nodes | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/serviceless-external-nodes.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/smart-update.log) |
| split-horizon | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/split-horizon.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/storage.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/tls-issue-cert-manager.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/upgrade.log) |
| upgrade-consistency | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/upgrade-consistency.log) |
| upgrade-consistency-sharded-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/upgrade-consistency-sharded-tls.log) |
| upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/upgrade-sharded.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/12dcce17/version-service.log) |
| We run 53 out of 53|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1733/commits/12dcce17b4f1a8db38d3d41006aa0935106a331c
image: `perconalab/percona-server-mongodb-operator:PR-1733-12dcce17`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/arbiter.log) |
| balancer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/balancer.log) |
| custom-replset-name | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/custom-replset-name.log) |
| custom-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/custom-tls.log) |
| custom-users-roles | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/custom-users-roles.log) |
| custom-users-roles-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/custom-users-roles-sharded.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/cross-site-sharded.log) |
| data-at-rest-encryption | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/data-at-rest-encryption.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/data-sharded.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/demand-backup.log) |
| demand-backup-fs | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/demand-backup-fs.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/demand-backup-eks-credentials.log) |
| demand-backup-physical | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/demand-backup-physical.log) |
| demand-backup-physical-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/demand-backup-physical-sharded.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/demand-backup-sharded.log) |
| expose-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/expose-sharded.log) |
| ignore-labels-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/ignore-labels-annotations.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/init-deploy.log) |
| finalizer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/finalizer.log) |
| ldap | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/ldap.log) |
| ldap-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/ldap-tls.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/limits.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/liveness.log) |
| mongod-major-upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/mongod-major-upgrade.log) |
| mongod-major-upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/mongod-major-upgrade-sharded.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/monitoring-2-0.log) |
| multi-cluster-service | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/multi-cluster-service.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/non-voting.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/one-pod.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/operator-self-healing-chaos.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/pitr.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/pitr-sharded.log) |
| pitr-physical | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/pitr-physical.log) |
| pvc-resize | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/pvc-resize.log) |
| recover-no-primary | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/recover-no-primary.log) |
| replset-overrides | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/replset-overrides.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/rs-shard-migration.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/scheduled-backup.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/security-context.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/self-healing-chaos.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/service-per-pod.log) |
| serviceless-external-nodes | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/serviceless-external-nodes.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/smart-update.log) |
| split-horizon | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/split-horizon.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/storage.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/tls-issue-cert-manager.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/upgrade.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/upgrade-consistency.log) |
| upgrade-consistency-sharded-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/upgrade-consistency-sharded-tls.log) |
| upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/upgrade-sharded.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1733/bb9bd4de/version-service.log) |
| We run 53 out of 53|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1733/commits/bb9bd4de002c9d5741c90099d0c0759e14276c23
image: `perconalab/percona-server-mongodb-operator:PR-1733-bb9bd4de`

**Pull request closed without merging**